### PR TITLE
[ONNX] Filter out torchscript sentences

### LINKED ISSUE #157300: PyTorch Tutorial Audit - ONNX
### 📚 The doc issue

This issue is created because the following doc pages have been marked as containing mentions of/using one of the outdated PyTorch technologies:

https://github.com/pytorch/pytorch/blob/main/docs/source/onnx_dynamo_memory_usage.rst
https://github.com/pytorch/pytorch/blob/main/docs/source/onnx_torchscript_supported_aten_ops.rst#L3
https://github.com/pytorch/pytorch/blob/main/docs/source/onnx_dynamo.rst
https://github.com/pytorch/pytorch/blob/main/docs/source/onnx.rst

Follow this checklist to address the issues in this page:

- [ ] If page is irrelevant, delete it and ignore below checkboxes
- [ ] Verify that none of the following outdated technologies are present in this doc:
  - [ ] FSDP1
  - [ ] TorchScript
  - [ ] Torchserve
  - [ ] FX tracing
  - [ ] Old TorchMobile (all should reference ExecuTorch)
- [ ] The doc does not reference old versions of PyTorch except to reference the minimum required version and is free of phrases like "In PyTorch 1.8".
- [ ] Add the module:docs label to the PR 

If we don't see a pull request by July 31, we will start deprecating the above mentioned tutorials.

### Suggest a potential alternative/fix

_No response_

cc @justinchuby @svekars @AlannaBurke

### PR DESCRIPTION
**Pull request by titaiwangms**
Fixes #157300 

cc @justinchuby

### CODE REVIEW

File: docs/source/onnx.md [hunk 1]
```diff
@@ -40,11 +40,9 @@ torch.onnx.export(
 )
 ```
 
-Next sections introduce the two versions of the exporter.
-
 ## TorchDynamo-based ONNX Exporter
 
-*The TorchDynamo-based ONNX exporter is the newest (and Beta) exporter for PyTorch 2.1 and newer*
+*The TorchDynamo-based ONNX exporter is the newest exporter for PyTorch 2.1 and newer*
```
Comment (justinchuby) [inline docs/source/onnx.md]:
```suggestion
*The TorchDynamo-based ONNX exporter is the newest exporter for PyTorch 2.6 and newer*
```

File: docs/source/onnx.md [hunk 2]
```diff
@@ -40,11 +40,9 @@ torch.onnx.export(
 )
 ```
 
-Next sections introduce the two versions of the exporter.
-
 ## TorchDynamo-based ONNX Exporter
```
Comment (justinchuby) [inline docs/source/onnx.md]: Can you update the reference to say `torch.export` based?

File: docs/source/onnx_dynamo_memory_usage.rst
Comment (justinchuby) [inline docs/source/onnx_dynamo_memory_usage.rst:1]: I don't think this page is important. I would consider removing it

File: docs/source/onnx_torchscript_supported_aten_ops.rst
Comment (justinchuby) [inline docs/source/onnx_torchscript_supported_aten_ops.rst:1]: We can remove build_onnx_torchscript_supported_aten_op_csv_table as well

**Review by justinchuby — APPROVED** [review]

File: docs/source/onnx.md (hunk 1 above)
Comment (justinchuby) [reply docs/source/onnx.md]: Prior to that the export was unusable. So consider updating our statement

File: docs/source/onnx.md (hunk 2 above)
Comment (titaiwangms) [reply docs/source/onnx.md]: I thought about this. Do we want to directly use `torch.export.export` (API name) to describe it, or there is a better way?

Comment (justinchuby) [reply docs/source/onnx.md]: I think the export team suggested "PT2 export" previously. @angelayi do you have suggestions? Thanks

File: docs/source/onnx_torchscript_supported_aten_ops.rst
Comment (titaiwangms) [reply docs/source/onnx_torchscript_supported_aten_ops.rst:1]: Done

File: docs/source/onnx_dynamo_memory_usage.rst
Comment (titaiwangms) [reply docs/source/onnx_dynamo_memory_usage.rst:1]: Removed

File: docs/source/onnx.md (hunk 2 above)
Comment (titaiwangms) [reply docs/source/onnx.md]: Changed to pt2 and PyTorch2

**Comment by titaiwangms** [comment]
Suppressed bc linter as `docs/source/scripts/onnx/build_onnx_torchscript_supported_aten_op_csv_table.py` is not a public API.

File: docs/source/onnx_pt2.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# TorchDynamo-based ONNX Exporter
+# PyTorch2-based ONNX Exporter
```
Comment (svekars) [inline docs/source/onnx_pt2.md]: I think using Dynamo is okay? Torchscript was the main concern

File: docs/source/onnx.md [hunk 4]
```diff
@@ -40,39 +38,18 @@ torch.onnx.export(
...
 
-*The TorchDynamo-based ONNX exporter is the newest (and Beta) exporter for PyTorch 2.1 and newer*
-
-TorchDynamo engine is leveraged to hook into Python's frame evaluation API and dynamically rewrite its
+PyTorch 2.0 engine is leveraged to hook into Python's frame evaluation API and dynamically rewrite its
 bytecode into an FX Graph. The resulting FX Graph is then polished before it is finally translated into an
 ONNX graph.
 
 The main advantage of this approach is that the [FX graph](https://pytorch.org/docs/stable/fx.html) is captured using
 bytecode analysis that preserves the dynamic nature of the model instead of using traditional static tracing techniques.
```
Comment (justinchuby) [inline docs/source/onnx.md]: Looks like this would not be accurate for non-strict? Maybe we can just link to torch.export docs.

File: docs/source/onnx_pt2.md (hunk 3 above)
Comment (titaiwangms) [reply docs/source/onnx_pt2.md]: The issue is that under the hood, ONNX exporter is not really using dynamo, instead, we use `torch.export.export(..., strict=False)` as our upstream. The documentation was wrong (even `dynamo` is used in `torch.onnx.export` API) due to legacy reasons. It could be bc and huge impact if we rename `dynamo` in our API, so we decided to keep it, since it's going to be deleted anyway (as torchscript EOL). 

But in the docs, at least we want to keep the correct information/knowledge to the users what's really going on behind the API.

File: docs/source/onnx.md (hunk 4 above)
Comment (angelayi) [reply docs/source/onnx.md]: Yup that's correct.

> Maybe we can just link to torch.export docs.

What are your opinions on calling it torch.export based onnx exporter?

Comment (justinchuby) [reply docs/source/onnx.md]: That works - whatever the torch.export team recommends :)

File: docs/source/onnx.md [hunk 5]
```diff
@@ -88,3 +88,29 @@ for tracking purposes -->
 .. py:module:: torch.onnx.symbolic_opset9
 .. py:module:: torch.onnx.utils
 ```
+
+Python API
+----------
+
```
Comment (justinchuby) [inline docs/source/onnx.md]: Use markdown

File: docs/source/onnx.md [hunk 6]
```diff
@@ -88,3 +88,29 @@ for tracking purposes -->
...
+.. automodule:: torch.onnx
```
Comment (justinchuby) [inline docs/source/onnx.md]: Use `{eval-rst}`

https://myst-parser.readthedocs.io/en/latest/faq/index.html#include-rst-files-into-a-markdown-file

File: docs/source/onnx_pt2.md
Comment (justinchuby) [inline docs/source/onnx_pt2.md:1]: maybe rename the file to onnx_export ?

File: docs/source/onnx.md [hunk 7]
```diff
@@ -88,3 +88,29 @@ for tracking purposes -->
...
+
+Functions
+^^^^^^^^^
+
+.. autofunction:: export
+    :noindex:
+.. autofunction:: register_custom_op_symbolic
+.. autofunction:: unregister_custom_op_symbolic
+.. autofunction:: select_model_mode_for_export
```
Comment (justinchuby) [inline docs/source/onnx.md]: These functions are obsolete. We can move them to the section above which will store a reference but will not display them

Comment (justinchuby) [reply docs/source/onnx.md]: "This module needs to be documented. Adding here in the meantime
for tracking purposes" above

File: docs/source/onnx.md [hunk 8]
```diff
@@ -88,3 +88,29 @@ for tracking purposes -->
...
+
+Classes
+^^^^^^^
+
+.. autosummary::
+    :toctree: generated
+    :nosignatures:
+    :template: classtemplate.rst
+
+    JitScalarType
```
Comment (justinchuby) [inline docs/source/onnx.md]: This one can be hidden too

File: docs/source/onnx_pt2.md
Comment (titaiwangms) [reply docs/source/onnx_pt2.md:1]: Done

File: docs/source/onnx.md (hunk 7 above)
Comment (titaiwangms) [reply docs/source/onnx.md]: Done

File: docs/source/onnx.md (hunk 8 above)
Comment (titaiwangms) [reply docs/source/onnx.md]: Done

File: docs/source/onnx.md [hunk 9]
```diff
@@ -42,26 +42,67 @@ torch.onnx.export(
...
+.. collapse:: Deprecated API (click to expand)
+   :class: warning
+
+   .. deprecated:: 2.6
+      These functions are deprecated and will be removed in a future version.
+
+   .. autofunction:: register_custom_op_symbolic
+   .. autofunction:: unregister_custom_op_symbolic
+   .. autofunction:: select_model_mode_for_export
+   .. autofunction:: is_in_onnx_export
```
Comment (justinchuby) [inline docs/source/onnx.md]: This is not deprecated

File: docs/source/onnx.md [hunk 10]
```diff
@@ -42,26 +42,67 @@ torch.onnx.export(
...
+  See {ref}`torch.cond <cond>`.
 
 ## Contributing / Developing
 
 The ONNX exporter is a community project and we welcome contributions. We follow the
 [PyTorch guidelines for contributions](https://github.com/pytorch/pytorch/blob/main/CONTRIBUTING.md), but you might
 also be interested in reading our [development wiki](https://github.com/pytorch/pytorch/wiki/PyTorch-ONNX-exporter).
 
+
+## Python API
```
Comment (justinchuby) [inline docs/source/onnx.md]: Everything is Python in the PyTorch Python documentation so the title feels redundant

**Review by svekars — APPROVED** [review]

File: docs/source/onnx.md [hunk 11]
```diff
@@ -40,54 +38,72 @@ torch.onnx.export(
...
+
+.. autofunction:: export
+    :noindex:
+
+### Deprecated API
+
+```{eval-rst}
+.. deprecated:: 2.6
+    These functions are deprecated and will be removed in a future version.
+
```
Comment (svekars) [inline docs/source/onnx.md:103]: the functions below need a .. currentmodule:: directive
